clang/test/dpct/helper_files_ref/include/rng_utils.hpp

```diff
@@ -174,6 +176,235 @@ template <typename engine_t> class rng_generator {
 };
 
 } // namespace device
+
+namespace host {
+enum class random_engine_type {
+  philox4x32x10,
+  mrg32k3a,
+  mt2203,
+  mt19937,
+  sobol
+};
+
+/// The random number generator on host.
+class rng_host {
+public:
+int create_generator(const random_engine_type &type) {
```

Comment:
1. is there a way to use a constructor instead?
2. made the name shorter
3. how the return value is supposed to be used?
```suggestion
int create(const random_engine_type &type) {
```

Reply:
1. Because user can call `curandCreateGenerator` of one generator many times, we need support calling the create function repeatly. So I think use a constructor to create a new object is not efficient. (but both implementation can work correctly)
2. fixed
3. CUDA will return a Status for each API, so I return a int value instead. I will delete the return value now, and migrated it to (*, 0) when the return value is used.